ccl/backupccl: TestBackupExportRequestTimeout failed

ccl/backupccl.TestBackupExportRequestTimeout [failed](https://tanzanite.cluster.engflow.com/invocations/default/80bb9f91-6da9-46ee-b67b-5542be24b843?testReportRun=3&testReportShard=46&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.2 @ [742a2251a05088a238fc6eb4c2466dace3cb1bb2](https://github.com/cockroachdb/cockroach/commits/742a2251a05088a238fc6eb4c2466dace3cb1bb2):


```
        testing.runTests(0xc002405e90, {0xc00342b5c0, 0x6, 0x8}, {0xc0000061c0?, 0xc002405e30?, 0x12b148e0?})
        	GOROOT/src/testing/testing.go:2159 +0x8bf
        testing.(*M).Run(0xc003480000)
        	GOROOT/src/testing/testing.go:2027 +0xf18
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test.TestMain(0xc003480000)
        	github.com/cockroachdb/cockroach/pkg/ccl/backupccl_test/pkg/ccl/backupccl/main_test.go:31 +0x1bb
        main.main()
        	main/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/testmain.go:736 +0x825
        goroutine 5 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc000f99a00)
        	go.opencensus.io/stats/view/external/io_opencensus_go/stats/view/worker.go:292 +0x1a5
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	go.opencensus.io/stats/view/external/io_opencensus_go/stats/view/worker.go:34 +0xfd
        goroutine 37 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:172 +0x127
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/goschedstats/runnable.go:164 +0x25
        goroutine 7 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:128 +0x6e
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:83 +0x25
        goroutine 8 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:152 +0xae
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:84 +0x31
        goroutine 51 [syscall]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x29
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1d
        created by os/signal.Notify.func1.1 in goroutine 8
        	GOROOT/src/os/signal/signal.go:151 +0x47
        goroutine 21 [chan receive]:
        github.com/bazelbuild/rules_go/go/tools/bzltestutil.RegisterTimeoutHandler.func1()
        	github.com/bazelbuild/rules_go/go/tools/bzltestutil/external/io_bazel_rules_go/go/tools/bzltestutil/timeout.go:34 +0x31
        created by github.com/bazelbuild/rules_go/go/tools/bzltestutil.RegisterTimeoutHandler in goroutine 1
        	github.com/bazelbuild/rules_go/go/tools/bzltestutil/external/io_bazel_rules_go/go/tools/bzltestutil/timeout.go:33 +0xe5
        goroutine 15013 [chan receive]:
        github.com/cockroachdb/pebble/sstable.(*writeQueue).runWorker(0xc0117667c0)
        	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/write_queue.go:95 +0x65
        created by github.com/cockroachdb/pebble/sstable.newWriteQueue in goroutine 15003
        	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/write_queue.go:57 +0x185: Leaked goroutine: goroutine 15013 [chan receive]:
        github.com/cockroachdb/pebble/sstable.(*writeQueue).runWorker(0xc0117667c0)
        	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/write_queue.go:95 +0x65
        created by github.com/cockroachdb/pebble/sstable.newWriteQueue in goroutine 15003
        	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/write_queue.go:57 +0x185
--- FAIL: TestBackupExportRequestTimeout (21.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=46</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupExportRequestTimeout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40617

--- a/pkg/ccl/backupccl/file_sst_sink.go
+++ b/pkg/ccl/backupccl/file_sst_sink.go
@@ -100,6 +100,7 @@ func (s *fileSSTSink) Close() error {
 	if s.out != nil {
 		return s.out.Close()
 	}
+	s.sst.Close()
 	return nil
 }
 

